### ISSUE TITLE
[Feature Request] Add closing parenthesis ')' as a punctation sign marking the end of the shorthand form of cross reference UID

### ISSUE THREAD

**Issue by GlmGb** [dotnet/docfx#8439]
I often put UIDs of my API object in shorthand form in parenthesis, either to link to the object that the current sentence refers to, or to give examples:
Ex. "_this functionality is related to entities that support this feature (@Namespace1.Myobject1 and @Namespace1.Myobject2)_"

But since ')' is not a valid punctuation to mark the end of the UI, only @Namespace1.Myobject1 will be converted to a link to the matching object in API documentation, but @Namespace1.Myobject2 is left "as is":
"_this functionality is related to entities that support this feature ([Myobject1](#) and @Namespace1.Myobject2)_"

The current workaround is to have an extra space ' ' before the closing parenthesis, or use the more complicated markdown syntax, but it is neither intuitive nor elegant.

I expect a closing parenthesis to be considered as a sign that marks the end of a UID (like ``[.,;:!?`~]``), in the context where this closing parenthesis has no other function than a standard text sign.

**Comment by GlmGb** [comment]
Yes, `@System.Object.Equals(System.Object)` works as expected, but as you mentioned, the closing parenthesis is in this case part of the UID, the parser is able to know it since there is a matching opening parenthesis after the '@'.

**Comment by yufeih** [comment]
Does quote work _(@"Namespace1.Myobject1" and @"Namespace1.Myobject2")_?
